alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexWorkflowAction.kt

```diff
@@ -0,0 +1,562 @@
+import org.opensearch.commons.alerting.model.ScheduledJob
+import org.opensearch.commons.alerting.model.ScheduledJob.Companion.SCHEDULED_JOBS_INDEX
+import org.opensearch.commons.alerting.model.Workflow
+import org.opensearch.commons.authuser.User
+import org.opensearch.commons.utils.recreateObject
+import org.opensearch.index.IndexNotFoundException
+import org.opensearch.index.query.QueryBuilders
+import org.opensearch.rest.RestRequest
+import org.opensearch.rest.RestStatus
+import org.opensearch.search.builder.SearchSourceBuilder
+import org.opensearch.tasks.Task
+import org.opensearch.transport.TransportService
+import java.util.stream.Collectors
+
+private val log = LogManager.getLogger(TransportIndexWorkflowAction::class.java)
+private val scope: CoroutineScope = CoroutineScope(Dispatchers.IO)
+
+class TransportIndexWorkflowAction @Inject constructor(
+    transportService: TransportService,
+    val client: Client,
+    actionFilters: ActionFilters,
+    val scheduledJobIndices: ScheduledJobIndices,
+    val clusterService: ClusterService,
+    val settings: Settings,
+    val xContentRegistry: NamedXContentRegistry,
+    val namedWriteableRegistry: NamedWriteableRegistry,
+) : HandledTransportAction<ActionRequest, IndexWorkflowResponse>(
+    AlertingActions.INDEX_WORKFLOW_ACTION_NAME, transportService, actionFilters, ::IndexWorkflowRequest
+),
+    SecureTransportAction {
+
+    @Volatile
+    private var maxMonitors = ALERTING_MAX_MONITORS.get(settings)
+
+    @Volatile
+    private var requestTimeout = REQUEST_TIMEOUT.get(settings)
+
+    @Volatile
+    private var indexTimeout = INDEX_TIMEOUT.get(settings)
+
+    @Volatile
+    private var maxActionThrottle = MAX_ACTION_THROTTLE_VALUE.get(settings)
+
+    @Volatile
+    private var allowList = ALLOW_LIST.get(settings)
+
+    @Volatile
+    override var filterByEnabled = AlertingSettings.FILTER_BY_BACKEND_ROLES.get(settings)
+
+    init {
+        clusterService.clusterSettings.addSettingsUpdateConsumer(ALERTING_MAX_MONITORS) { maxMonitors = it }
+        clusterService.clusterSettings.addSettingsUpdateConsumer(REQUEST_TIMEOUT) { requestTimeout = it }
+        clusterService.clusterSettings.addSettingsUpdateConsumer(INDEX_TIMEOUT) { indexTimeout = it }
+        clusterService.clusterSettings.addSettingsUpdateConsumer(MAX_ACTION_THROTTLE_VALUE) { maxActionThrottle = it }
+        clusterService.clusterSettings.addSettingsUpdateConsumer(ALLOW_LIST) { allowList = it }
+        listenFilterBySettingChange(clusterService)
+    }
+
+    override fun doExecute(task: Task, request: ActionRequest, actionListener: ActionListener<IndexWorkflowResponse>) {
+        val transformedRequest = request as? IndexWorkflowRequest
+            ?: recreateObject(request, namedWriteableRegistry) {
+                IndexWorkflowRequest(it)
+            }
+
+        val user = readUserFromThreadContext(client)
+
+        if (!validateUserBackendRoles(user, actionListener)) {
```

Comment:
can we also validate if the user has EXECUTE permission on the delegate monitors?